sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -851,9 +855,11 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
 
   private[this] def castToIntCode(from: DataType, ctx: CodegenContext): CastFunction = from match {
     case StringType =>
-      val wrapper = ctx.freshName("wrapper")
-      ctx.addMutableState("UTF8String.IntWrapper", wrapper,
-        s"$wrapper = new UTF8String.IntWrapper();")
+      val wrapper = "intWrapper"
+      if (!ctx.mutableStates.exists(s => s._1 == wrapper)) {
+        ctx.addMutableState("UTF8String.IntWrapper", wrapper,
+          s"$wrapper = new UTF8String.IntWrapper();")
+      }
       (c, evPrim, evNull) =>
         s"""
           if ($c.toInt($wrapper)) {
```

Comment:
what if we create a new wrapper every time?

Reply:
It would work well, too. We may have some overhead to create a small object `IntWrapper` and collect it at GC.

Reply:
shouldn't GC work very well in this case? My concern is, if there is no significant performance overhead, reusing global variables may be too hacky.

Reply:
I see. If we do not reuse global variables, we can use only a local variable.